oktapam/client/secret_folder.go

```diff
@@ -215,7 +233,11 @@ func ConvertSecretOrFolderListResponseToSecretFolder(resp pam.SecretOrFolderList
 	}
 }
 
-func ConvertResolveSecretOrFolderResponse(resp *pam.ResolveSecretOrFolderResponse) SecretFolder {
+func ConvertResolveSecretOrFolderResponseToSecretFolder(resp *pam.ResolveSecretOrFolderResponse) (SecretFolder, error) {
+	if resp.Type != nil && *resp.Type != pam.SecretType_FOLDER {
+		return SecretFolder{}, fmt.Errorf("resolved entity was not a secret folder.  id: %s", resp.Id)
+	}
+
 	var parentSecretFolderID *string
 
 	if len(resp.Path) != 0 {
```

Comment:
if we get down here and len is 0, that means it is a root level folder?

Reply:
that's correct